src/apns_connection.erl

```diff
@@ -251,10 +251,17 @@ build_payload(#apns_msg{alert = Alert,
                         badge = Badge,
                         sound = Sound,
                         apns_extra=Apns_Extra,
-                        extra = Extra}) ->
-    build_payload([{alert, Alert},
-                   {badge, Badge},
-                   {sound, Sound}] ++ Apns_Extra, Extra).
+                        extra = Extra} = Msg) ->
+    if is_binary(Sound) -> 
```

Comment:
when I see an `if` I always think _it should be wrong_, maybe we can find a better solution à la Erlang :-)